services/credential-issuance-server/src/agent.ts

```diff
@@ -40,17 +46,51 @@ class Agent {
 
   async issueAcdcCredentialByAid(schemaSaid, aid, attribute) {
     return this.signifyApi.issueCredential(
-      Agent.ISSUER_AID_NAME,
+      Agent.HOLDER_AID_NAME,
       this.keriRegistryRegk,
       schemaSaid,
       aid,
       attribute
     );
   }
+  async issueLeChainedCredential(legalEntityAidPrefix: string) {
+    const AIDIssuerName = Agent.ISSUER_AID_NAME;
+    const AIDHolderName = Agent.HOLDER_AID_NAME;
+    const issuerAid = await this.signifyApiIssuer.getIdentifierByName(AIDIssuerName);
+    const holderAid = await this.signifyApi.getIdentifierByName(AIDHolderName);
+    const issuerAidOobi = await this.signifyApiIssuer.getOobi(issuerAid.name);
+    const holderAidOobi = await this.signifyApi.getOobi(holderAid.name);
+    await this.signifyApi.resolveOobi(issuerAidOobi);
+    await this.signifyApiIssuer.resolveOobi(holderAidOobi);
+    // Qvi
+    const qviCredentialId = await this.signifyApiIssuer.issueQVICredential(
+      issuerAid.name,
+      this.keriIssuerRegistryRegk,
+      "EBfdlu8R27Fbx-ehrqwImnK-8Cm79sqbAQ4MmvEAYqao",
+      holderAid.prefix
+    );
+    await new Promise((rs) => {
+      setTimeout(() => {
+        rs(true);
+      }, 5000);
+    });
+    const holderNotifications = await this.signifyApi.getNotifications();
+    const grantNotification = holderNotifications.notes[0];
```

Comment:
This needs to be more robust rather than assuming the most recent notification is this grant. We might have many people calling the server in parallel.

Reply:
This still hasn't been addressed - we are still assuming the 0th notification is the correct one.

If it's too complex to do, then perhaps instead we can just issue 1 QVI credential on start-up immediately, and chain LE credentials to this one always...

Otherwise I can see this getting really buggy when we assume the 0th notification is the right one as we might have parallel requests

Reply:
Realistically just creating 1 QVI holder on start-up is the cleaner option I think.

Then whenever we issue a credential, if it's an LE credential, chain the QVI credential onto it.

If possible, we should make sure the API is not accessible until after the QVI credential has been initialised - not sure how easy this is